commcid.go

```diff
@@ -129,18 +83,18 @@ func CIDToDataCommitmentV1(c cid.Cid) ([]byte, error) {
 	if err != nil {
 		return nil, err
 	}
-	if hash != FC_UNSEALED_V1 {
+	if hash != cid.FilCommitmentUnsealed {
 		return nil, ErrIncorrectHash
 	}
 	return commD, nil
 }
 
 // ReplicaCommitmentV1ToCID converts a raw data commitment to a CID
 // by adding:
-// - serialization type of raw
-// - hashing type of Filecoin sealed hashing function v1 (0xfc2)
+// - codec of type FC_SEALED_V1
+// - hashing type of FC_SHA2_256_TRUNC254
 func ReplicaCommitmentV1ToCID(commR []byte) cid.Cid {
-	c, _ := CommitmentToCID(commR, FC_SEALED_V1)
+	c, _ := CommitmentToCID(commR, cid.FilCommitmentSealed, multihash.SHA2_256_TRUNC254_PADDED)
```

Comment:
iirc CommR uses sealed+Poseidon. CommD and CommP use unsealed + the special sha256. @porcuquine can you confirm please before we commit to these identifiers?

Reply:
That's right. CommR is the output of a binary Poseidon hash and has no relationship to SHA256. Does that suffice to answer your question?

Reply:
@porcuquine yep, thanks!

@ribasushi so, better change this to `multihash. POSEIDON_BLS12_381_A1_FC1` and the comment above it too

Reply:
Offfff thanks for catching this folks! I misread an earlier comment to arrive at this 🤦 
All fixed!

Also rewrote the validator from scratch to disallow incorrect codec/hash combinations. Namely:
https://github.com/filecoin-project/go-fil-commcid/blob/226ec33a83e9c5cd6d01029cbff44590e70fe92c/commcid.go#L111-L133